server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -154,10 +156,18 @@ protected void doExecute(Task task, BulkRequest bulkRequest, ActionListener<Bulk
         final Releasable releasable = indexingPressure.markCoordinatingOperationStarted(indexingOps, indexingBytes, isOnlySystem);
         final ActionListener<BulkResponse> releasingListener = ActionListener.runBefore(listener, releasable::close);
         final String executorName = isOnlySystem ? Names.SYSTEM_WRITE : Names.WRITE;
-        try {
-            doInternalExecute(task, bulkRequest, executorName, releasingListener);
-        } catch (Exception e) {
-            releasingListener.onFailure(e);
+        ActionRunnable<BulkResponse> internalExecute = new ActionRunnable<>(releasingListener) {
+            @Override
+            protected void doRun() {
+                doInternalExecute(task, bulkRequest, executorName, releasingListener);
+            }
+        };
+        // We dispatch large bulk requests as coordinating these bytes can occur on the transport thread and might involve
+        // compression.
+        if (indexingBytes > ONE_MEGABYTE) {
```

Comment:
Perhaps `>=`? Feels more natural but again optional. Otherwise we are saying that a request of size `1MB+1` is a large request.